### PR TITLE
refactor: [#2] api client

### PR DESCRIPTION
**Pull request by gjivaros**
### Modify the **request** method so as not to modify parameters inside the method. 

**Related issue :** #2

### CODE REVIEW

**Comment by gjivaros** [comment]
@koladev32 
@LewisYann

**Comment by koladev32** [comment]
Hi @gjivaros 

It will be nice if this PR is linked to an issue.

File: src/apiClient.ts [hunk 1]
```diff
@@ -34,34 +34,35 @@ export class TransfaAPIClient {
...
       ...headers,
     };
 
-    if (data !== undefined) {
-      data = JSON.stringify(data);
-      headers["content-type"] = "application/json;charset=utf-8";
+    /**
+     * can't do if(requestPayload) because requestPayload can be equal to 0 😃
+     */
+    if (requestPayload !== undefined || requestPayload !== null) {
```
Comment (LewisYann) [inline src/apiClient.ts]:
```suggestion
    if (requestPayload !== undefined && requestPayload !== null) {
```

Comment (LewisYann) [reply src/apiClient.ts]: We need to ensure that `requestPayload` has a valid value

**Review by LewisYann — CHANGES_REQUESTED** [review]

File: src/apiClient.ts [hunk 2]
```diff
@@ -34,34 +34,35 @@ export class TransfaAPIClient {
...
     data?: any,
-    params?: any,
+    params: Record<string, any> = {},
     headers?: AxiosRequestConfig["headers"]
   ): Promise<AxiosResponse<T>> {
-    if (!params) {
-      params = {};
-    }
 
+    let requestPayload = data
```
Comment (LewisYann) [inline src/apiClient.ts:41]: This line don't prevent the update of the `data` object.

Use the spread operator to clone the object, then verify that the cloning has indeed occurred and that the cloned item is indeed an object.

**Comment by gjivaros** [comment]
Someone can be create an issue for this refactoring ? @LewisYann ☺️

**Review by LewisYann — CHANGES_REQUESTED** [review]

File: src/apiClient.ts (hunk 2 above)
Comment (gjivaros) [reply src/apiClient.ts:41]: Good point @LewisYann . But data is of type any

File: src/apiClient.ts (hunk 1 above)
Comment (gjivaros) [reply src/apiClient.ts]: Good point

File: src/apiClient.ts (hunk 2 above)
Comment (LewisYann) [reply src/apiClient.ts:41]: I will improve the typing later

File: src/apiClient.ts (hunk 1 above)
Comment (gjivaros) [reply src/apiClient.ts]: I do this correction @LewisYann . Thanks for the review

Comment (LewisYann) [reply src/apiClient.ts]: Merci à toi @gjivaros

**Review by LewisYann — APPROVED** [review]

**Pull request merged**